### PR TITLE
[GMDX-399] Adding account configuration page

### PR DESCRIPTION
**Pull request by GolanSG**

### CODE REVIEW

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
+//
+//  AccountDetailsViewController.swift
+//  GCMessengerSDKSample
+//
+//  Created by Golan Shoval Gil on 26/06/2022.
+//
```
Comment (ElizaSapir) [inline GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:6]: Please remove lines 1-6 and add below:

```
// ===================================================================================================
// Copyright © 2022 GenesysCloud(Genesys).
// GenesysCloud SDK.
// All rights reserved.
// ===================================================================================================
```

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        view.endEditing(true)
+    }
+
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
+    
+    private func setupFields() {
+        deploymentIdTextField.placeholder = "Deployment id"
```
Comment (ElizaSapir) [inline GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:39]: can we add placeholders inside storyborad?

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        
+        tokenTextField.placeholder = "Token store id"
+        tokenTextField.text = UserDefaults.token
+        
+        loggingSwitch.setOn(UserDefaults.logging, animated: true)
+    }
+
+    @IBAction func startChatButtonTapped(_ sender: UIButton) {
+        
+        let account = MessengerAccount(deploymentId: deploymentIdTextField.text ?? "",
```
Comment (ElizaSapir) [inline GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:53]: Is the tap ignored when the fields are empty?

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                                       domain: domainIdTextField.text ?? "",
+                                       tokenStoreKey: tokenTextField.text ?? "",
+                                       logging: loggingSwitch.isOn)
+        
+        updateUserDefaults()
+        openMainController(with: account)
```
Comment (ElizaSapir) [inline GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:59]: Should be opened when you get "ChatStarted" event, please follow below:

https://genesys.github.io/bold360-mobile-docs-ios/docs/chat-configuration/tracking-events/chat-lifecycle/#overview

**Review by ElizaSapir — CHANGES_REQUESTED** [review]

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift (hunk 3 above)
Comment (GolanSG) [reply GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:53]: Not on the current implementation, so all fields are required?

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift (hunk 4 above)
Comment (GolanSG) [reply GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift:59]: I'm not really doing any chatViewController related stuff on this viewController, I'm taking params entered & moving them to the ChatWrapperViewController @ElizaSapir

File: GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift (hunk 3 above)
Comment (GolanSG) [reply GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift]: As agreed with @tehilaR2 , first 2 params are required, and now presented as such.

**Review by tehilaR2 — CHANGES_REQUESTED** [review]

**Review by tehilaR2 — APPROVED** [review]

File: GCMessengerSDKSample/GCMessengerSDKSample/Base.lproj/Main.storyboard [hunk 5]
```diff
@@ -1,24 +1,167 @@
...
+                                                        <rect key="frame" x="0.0" y="54" width="374" height="34"/>
+                                                        <fontDescription key="fontDescription" type="system" pointSize="14"/>
+                                                        <textInputTraits key="textInputTraits" returnKeyType="next"/>
+                                                        <userDefinedRuntimeAttributes>
+                                                            <userDefinedRuntimeAttribute type="string" keyPath="_placeholder" value="Domain ID"/>
+                                                        </userDefinedRuntimeAttributes>
+                                                    </textField>
+                                                    <textField opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="left" contentVerticalAlignment="center" borderStyle="roundedRect" placeholder="Token store id" textAlignment="natural" minimumFontSize="17" translatesAutoresizingMaskIntoConstraints="NO" id="67a-ep-PGK">
```
[the remark this answers is not in the document]
Comment (GolanSG) [reply GCMessengerSDKSample/GCMessengerSDKSample/Base.lproj/Main.storyboard]: simulator_screenshot_3DB293D9-23E8-4941-BD38-7112DFA57EB7

File: GCMessengerSDKSample/GCMessengerSDKSample/Extensions/UITextField+Extensions.swift [hunk 6]
```diff
@@ -0,0 +1,28 @@
+//
+//  UITextField+Extensions.swift
+//  GCMessengerSDKSample
+//
+//  Created by Golan Shoval Gil on 29/06/2022.
+//
```
Comment (ElizaSapir) [inline GCMessengerSDKSample/GCMessengerSDKSample/Extensions/UITextField+Extensions.swift:6]: Please replace above lines with:

```
// ===================================================================================================
// Copyright © 2022 GenesysCloud(Genesys).
// GenesysCloud SDK.
// All rights reserved.
// ===================================================================================================
```

**Review by ElizaSapir — CHANGES_REQUESTED** [review]

**Review by ElizaSapir — APPROVED** [review]

**Pull request merged**